From 78265ef4686eb72f163802b63f7e145f96a88776 Mon Sep 17 00:00:00 2001
From: Wolfram Sang <w.sang@pengutronix.de>
Date: Thu, 28 Apr 2011 11:51:57 +0200
Subject: [PATCH] fbset: fix makefile

* do not overwrite CC to allow crosscompiling
* use DESTDIR
* don't create nodes or remove files in install

Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>
Signed-off-by: Wolfram Sang <w.sang@pengutronix.de>
---
 Makefile |   17 ++++-------------
 1 file changed, 4 insertions(+), 13 deletions(-)

Index: fbset-2.1/Makefile
===================================================================
--- fbset-2.1.orig/Makefile
+++ fbset-2.1/Makefile
@@ -2,7 +2,7 @@
 # Linux Frame Buffer Device Configuration
 #
 
-CC =		gcc -Wall -O2 -I.
+CFLAGS +=	-Wall -O2 -I.
 BISON =		bison -d
 FLEX =		flex
 INSTALL =	install
@@ -24,18 +24,9 @@ modes.tab.c:	modes.y
 		$(BISON) modes.y
 
 install:	fbset
-		if [ -f /sbin/fbset ]; then rm /sbin/fbset; fi
-		$(INSTALL) fbset /usr/sbin
-		$(INSTALL) fbset.8 /usr/man/man8
-		$(INSTALL) fb.modes.5 /usr/man/man5
-		if [ ! -c /dev/fb0 ]; then mknod /dev/fb0 c 29 0; fi
-		if [ ! -c /dev/fb1 ]; then mknod /dev/fb1 c 29 32; fi
-		if [ ! -c /dev/fb2 ]; then mknod /dev/fb2 c 29 64; fi
-		if [ ! -c /dev/fb3 ]; then mknod /dev/fb3 c 29 96; fi
-		if [ ! -c /dev/fb4 ]; then mknod /dev/fb4 c 29 128; fi
-		if [ ! -c /dev/fb5 ]; then mknod /dev/fb5 c 29 160; fi
-		if [ ! -c /dev/fb6 ]; then mknod /dev/fb6 c 29 192; fi
-		if [ ! -c /dev/fb7 ]; then mknod /dev/fb7 c 29 224; fi
+		$(INSTALL) -D fbset $(DESTDIR)/usr/sbin
+		$(INSTALL) -D fbset.8 $(DESTDIR)/usr/man/man8
+		$(INSTALL) -D fb.modes.5 $(DESTDIR)/usr/man/man5
 
 clean:
 		$(RM) *.o fbset lex.yy.c modes.tab.c modes.tab.h
